core/block_svg.js

```diff
@@ -150,9 +150,11 @@ Blockly.BlockSvg.prototype.select = function() {
     return;
   }
   if (Blockly.selected == this) {
-    return;
+    var oldId = this.id;
+  }
+  else {
+    var oldId = null;
   }
```

Comment:
in your version, every event goes through all the code after this line. Wouldn't it be sufficiant to simply avoid the return with:
`if (Blockly.selected !== this) {` ?